lib/process.js

```diff
@@ -25,12 +25,24 @@ exports.process = async ({ record, log, db, dynamo, ilp }) => {
   // Payout amount is in millicents (1/1000 of a cent)
   const payoutAmount = payouts.reduce((sum, p) => sum + p.amount, 0)
 
-  await ilp.sendIlpPayment({ pointer: ilpPaymentPointer, amount: payoutAmount })
+  const { success, remainingAmount } = await ilp.sendIlpPayment({ pointer: ilpPaymentPointer, amount: payoutAmount })
+  // If sending money for some reason didn't send the whole amount, mark this payout as paid and add a new
+  // payout that is the remaining amount
+  if (!success && remainingAmount > 0) {
+    await db.addDifferentialPayout({ maintainerId, remainingAmount })
+  }
 
   await db.updatePayoutsToPaid({ payoutIds, maintainerId, ilpPaymentPointer })
 
   await dynamo.unlockMaintainer({ maintainerId })
 
+  // If the ILP sender did not succeed in all senses of the word, then throw an error so we're notified and
+  // can dig in. It's fine to throw here, because even if this lambda fires again for the same maintainer,
+  // no payments could ever be duplicately paid.
+  if (!success) {
+    throw new Error('For some reason ILP sender failed')
+  }
```

Comment:
Can't we get an error in the return along with `success` and `remainingAmount` ? That way when we throw/log we can do so with the actual issue.